src/index.js

```diff
@@ -1,32 +1,43 @@
 var XMLHttpRequest = require("xmlhttprequest").XMLHttpRequest;
 
-var API = 'https://rickandmortyapi.com/api/character/';
+var API = "https://rickandmortyapi.com/api/character/";
+var info = [];
+
 var xhttp = new XMLHttpRequest();
 
-function fetchData(url_api, callback) {
-  xhttp.onreadystatechange = function (event) {
-    if (xhttp.readyState === '4') {
-      if (xhttp.status == 200)
-        callback(null, xhttp.responseText);
-      else return callback(url_api);
-    }
-  };
-  xhttp.open('GET', url_api, false);
-  xhttp.send();
+const fetchData = url_api => {
+  return new Promise((resolve, reject) => {
+    xhttp.onreadystatechange = event => {
+      if (xhttp.readyState === 4) {
+        if (xhttp.status === 200) {
+          resolve(JSON.parse(xhttp.responseText));
+        } else reject(null);
```

Comment:
Te recomiendo ser mas claros en el reject, esto te permite identificar el error mas fácil. 

```suggestion
          reject(new Error("Algo salió mal."));
```